include/broker/zeek.hh

```diff
@@ -105,6 +130,104 @@ public:
     return get<std::string>(get<vector>(as_vector()[2])[0]);
   }
 
+  /// Support iteration with structured binding.
+  class metadata_iterator {
+    using iterator_category = std::forward_iterator_tag;
+    using difference_type = std::ptrdiff_t;
+    using value_type = std::pair<count, const data&>;
+    using pointer = value_type*;
+    using reference = value_type&;
+
+  public:
+    metadata_iterator(const broker::data* ptr) : ptr_(ptr) {}
+
+    const std::pair<count, const data&> operator*() const {
+      auto entry_ptr = get_if<vector>(*ptr_);
+      BROKER_ASSERT(entry_ptr && entry_ptr->size() == 2);
+      const auto& entry = *entry_ptr;
+      return {get<count>(entry[0]), entry[1]};
+    }
+
+    metadata_iterator operator++() {
+      ++ptr_;
+      return *this;
+    }
+
+    metadata_iterator operator++(int) {
+      auto tmp = *this;
+      ++(*this);
+      return tmp;
+    }
+
+    bool operator!=(metadata_iterator& other) {
+      return ptr_ != other.ptr_;
+    }
+
+  private:
+    const broker::data* ptr_;
+  };
+
+  /// Supports iteration over metadata
+  class metadata_wrapper {
+  public:
+    metadata_wrapper(const vector* v) : v_(v) {}
+
+    metadata_iterator begin() const {
```

Comment:
Calling this without doing something with the iterator is pretty much guaranteed to be a mistake. Hence:

```suggestion
    [[nodiscard]] metadata_iterator begin() const noexcept {
```